app/scripts/modules/amazon/src/function/details/AmazonFunctionDetails.tsx

```diff
@@ -0,0 +1,142 @@
+export class AmazonFunctionDetails extends React.Component<IAmazonFunctionDetailsProps, any> {
+  constructor(props: IAmazonFunctionDetailsProps) {
+    super(props);
+    this.state = {
+      loading: true,
+    };
+  }
+
+  public extractFunction(): void {
+    const { app, functionObj: functionFromProps } = this.props;
+    const functionDef: IFunction = app.functions.data.find((test: IFunction) => {
+      return (
+        test.functionName === functionFromProps.functionName &&
+        test.region === functionFromProps.region &&
+        test.account === functionFromProps.account
+      );
+    });
+
+    if (functionDef) {
+      AwsReactInjector.functionReader
+        .getFunctionDetails('aws', functionFromProps.account, functionFromProps.region, functionFromProps.functionName)
+        .then((details: IAmazonFunctionSourceData[]) => {
+          if (details.length) {
+            this.setState({
+              functionDef: details[0] as IAmazonFunction,
+              loading: false,
+            });
+          }
+        });
+    } else {
+      this.setState({
+        functionDef: {},
+        loading: false,
+      });
+    }
+  }
+
+  public componentDidMount(): void {
+    const { app } = this.props;
+    const dataSource = app.functions;
+    dataSource.ready().then(() => {
+      const dataSourceUnsubscribe = dataSource.onRefresh(null, () => this.extractFunction());
+      this.setState({ dataSourceUnsubscribe });
+      this.extractFunction();
+    });
+  }
+
+  public componentWillUnmount() {
+    this.state.dataSourceUnsubscribe && this.state.dataSourceUnsubscribe();
+  }
+
+  public render() {
+    const { app } = this.props;
+    const { loading, functionDef } = this.state;
+    if (loading) {
+      // Don't bother computing any children if we're loading
+      return <Details loading={loading} />;
+    }
+
+    const functionDetails = (
+      <dl className="horizontal-when-filters-collapsed dl-horizontal dl-flex">
+        <dt>Last Modified </dt>
+        <dd>{functionDef.lastModified}</dd>
+        <dt>In</dt>
+        <dd>
+          <AccountTag account={functionDef.account} /> {functionDef.region}
+        </dd>
+        <dt>VPC</dt>
+        <dd>{functionDef.vpcConfig ? functionDef.vpcConfig.vpcId : 'Default'}</dd>
+        <dt>Function ARN</dt>
+        <dd>{functionDef.functionArn}</dd>
+        <dt>Revision ID</dt>
+        <dd>{functionDef.revisionId}</dd>
+        <dt>Version</dt>
+        <dd>{functionDef.version}</dd>
+        <dt>Event Source</dt>
+        <dd>
+          {functionDef.eventSourceMappings && functionDef.eventSourceMappings.length !== 0
+            ? functionDef.eventSourceMappings
+            : 'None'}
+        </dd>
+      </dl>
+    );
+
+    const functionDetailsSection = (
+      <CollapsibleSection heading="Function Details">{functionDetails}</CollapsibleSection>
+    );
+
+    return (
+      <Details loading={this.state.loading}>
+        {_.isEmpty(this.state.functionDef) ? (
```

Comment:
```suggestion
        {isEmpty(this.state.functionDef) ? (
```

Reply:
Done. Change will appear in the next push.